src/memory.js

```diff
@@ -1,18 +1,49 @@
 class MemoryGame {
   constructor(cards) {
-    this.cards = cards;
-    // add the rest of the class properties here
-  }
-
+    this.cards = cards
+    this.pickedCards = []
+    this.pairsClicked = 0
+    this.pairsGuessed = 0
+   }
+   
   shuffleCards() {
-    // ... write your code here
+    
+    if (!this.cards) {
+      return undefined
+    }
+  
+    
+    for (let i = this.cards.length -1; i > 0; i--) {
+      const randomShuffleCard = Math.floor(Math.random() * (i + 1))
+      this.cards[i] = this.cards[randomShuffleCard] 
+      this.cards[randomShuffleCard] = this.cards[i]
+  
+    }
+    return this.cards
   }
 
+
   checkIfPair(card1, card2) {
-    // ... write your code here
-  }
+    
+    this.pairsClicked += 1
+      if (card1 === card2) {
+        this.pairsGuessed += 1
+        return true
+      } else {
+        return false
+    } 
+  } 
 
-  checkIfFinished() {
-    // ... write your code here
+
+  
+checkIfFinished() {
+  
+  if (this.pairsGuessed === this.cards.length / 2) {
+    return true
+    } else {
+    return false
+    }
   }
```

Comment:
Si la respuesta es un booleano, puedes devolver lo que preguntas en el IF! haria lo mismo :D

```suggestion
checkIfFinished() {
  return this.pairsGuessed === this.cards.length / 2
  }
```